src/core/Icon/Icon.baseStyles.tsx

```diff
@@ -1,13 +1,31 @@
 import { css } from 'styled-components';
 
-export const iconBaseStyles = css`
-  display: inline-block;
-  vertical-align: baseline;
-
-  &.fi-icon--cursor-pointer {
-    cursor: pointer;
-    & * {
-      cursor: inherit;
+export const iconBaseStyles = ({
+  color,
+  fill,
+}: {
+  color?: string;
+  fill: string;
+}) => {
+  const resolvedColor = fill ?? color ?? 'currentColor';
+
+  return css`
+    display: inline-block;
+    vertical-align: baseline;
+
+    .fi-icon-base-fill {
+      fill: ${resolvedColor};
+    }
+
+    .fi-icon-base-stroke {
+      stroke: ${resolvedColor};
```

Comment:
I started to wonder what happens if there is no e.g. stroke defined for a given icon but color is given in props? Could this introduce problems by applying a stroke to something that was supposed to be "invisible"? One alternative could be to only apply stroke color when explicitly provided in props.

Reply:
My bad, this should work as expected since fi-icon-base-stroke is only present for paths that should be rendered. So all good here.

Reply:
Yeah, the class should only be there if the stroke has the same predefined color as the fill. I remember stopping to think about this as well for a bit when writing the logic 😄